docs/spark/how-to-guides/batch-processing.md

```diff
@@ -0,0 +1,124 @@
+
+1. Create a new .NET console app and add the Microsft.Spark NuGet package. Then, add the following using statements to your code.
+
+   ```csharp
+   using System;
+   using Microsoft.Spark.Sql;
+   using static Microsoft.Spark.Sql.Functions;
+   ```
+
+1. Add the following code to create a `s_referenceDate` variable and a `Run` function. The `Run` function takes the CSV file path as an argument.
+
+   ```csharp
+   // For later use when filtering based on date
+   static readonly DateTime s_referenceDate = new DateTime(2015 10, 20);
+
+   public void Run(string[] args)
+   {
+       if (args.Length != 1)
+       {
+            Console.Error.WriteLine(
+            "Usage: GitHubProjects <path to projects.csv>");
+            Environment.Exit(1);
+        }
+    }
+   ```
+
+1. Inside the `Run` function, add the following code to establish a new SparkSession. The SparkSession is the entry point to programming Spark with the Dataset and DataFrame API. By calling the `spark` object, you can access Spark and DataFrame functionality throughout your program.
+
+   ```csharp
+   SparkSession spark = SparkSession
+                .Builder()
+                .AppName("GitHub and Spark Batch")
+                .GetOrCreate();
+   ```
+
+1. Read the input file into a DataFrame, which is a distributed collection of data organized into named columns.
+
+   ```csharp
+   DataFrame projectsDf = spark
+       .Read()
+       .Schema("id INT, url STRING, owner_id INT, " +
+       "name STRING, descriptor STRING, language STRING, " +
+       "created_at STRING, forked_from INT, deleted STRING" +
+       "updated_at STRING")
+       .Csv(args[0]);
+
+   projectsDf.Show();
+   ```
+
+1. Use `DataFrame.Na()` to drop rows with NA (null) values, and the `DataFrameNaFunctions.Drop()` method to remove certain columns from your data. This helps prevent errors if you try to analyze null data or columns that are not relevant to your final analysis.
+
+   ```csharp
+   // Drop any rows with NA values
+   DataFrameNaFunctions dropEmptyProjects = projectsDf.Na();
+   DataFrame cleanedProjects = dropEmptyProjects.Drop("any");   
+   // Remove unnecessary columns
+   cleanedProjects = cleanedProjects.Drop("id", "url", "owner_id");
+   cleanedProjects.Show();
+   ```
+
+1. Spark SQL allows you to make SQL calls on your data. It's common to combine user-defined functions and Spark SQL so to apply a user-defined function to all rows of your DataFrame.
+
+   You can specifically call `spark.Sql` to mimic standard SQL calls seen in other types of apps. You can also call methods like `GroupBy` and `Agg` to specifically combine, filter, and perform calculations on your data.
+
+   ```csharp
+   // Average number of times each language has been forked
+   DataFrame groupedDF = cleanedProjects
+       .GroupBy("language")
+       .Agg(Avg(cleanedProjects["forked_from"])   
+   // Sort by most forked languages first
```

Comment:
Similar to above comment about code readability/dividing based on task, this snippet of code could benefit from more spacing.

```suggestion

   // Sort by most forked languages first
```